src/S3/S3Client.php

```diff
@@ -429,7 +429,11 @@ public function __construct(array $args)
      */
     public static function isBucketDnsCompatible($bucket)
     {
-        $bucketLen = strlen($bucket);
+        if (!is_string($bucket)) {
+            return false;
+        }
+
+        $bucketLen = $bucket !== null ? strlen($bucket) : 0;
```

Comment:
This check is not necessary if we are checking for is_string above; it will never be null and always be a string, right?